### ISSUE TITLE
iCloud3 v3 - General Questions & Help With Problems

### ISSUE THREAD

**Issue by gcobb321** [gcobb321/icloud3_v3#129]
Use this issue to ask for help with problems. Only v3 questions and problems will be answered here. If you are having a problem or v2 related question or problem, go [here](https://community.home-assistant.io/t/icloud3-v2-4-an-advanced-device-tracker-custom-component-that-is-integrated-with-the-ios-app-location-monitor/267554/819).

**Comment by chintito4ever** [comment]
Spent last two evenings setting this up but failed. It wouldn't let me add a device. Whenever I try to submit in "Update Tracked iCloud3 Device" it auto scrolls up to icloud device tracker identity. Doesn't show any error. I could only add my device, none of the my friendsNFamily member's phone being added. 
![image](https://github.com/gcobb321/icloud3_v3/assets/41648578/3d90de4f-6efa-48e3-9e39-4bd6406848d5)

**Comment by chintito4ever** [comment]
I see it could be the issue with Apple's latest change that broke FnF updates.

**Comment by gcobb321** [comment]
You need to select the FamIly Sharing List Device and the iOS App device (if the iOS App is installed on it) to add ít. Find my Friends devices are not available due to the Apple change. 

A message should have been displayed when all the FamShr, FmF and iOS app devices are None.

**Comment by gcobb321** [comment]
I just uploaded beta 16c1 that fixes the problem where error messages in the Update Devices screen were not being displayed. I did net change the version number. It will still read b16c

**Comment by chintito4ever** [comment]
Do you know if that change by Apple is permanent or not? this basically makes it no use for me currently as I mostly track via FnF method.

**Comment by gcobb321** [comment]
I have no idea

**Comment by thiago-s-medeiros** [comment]
I'm not been able to get airpod battery pencentage... always 0%

**Comment by gcobb321** [comment]
Mine are always zero too. The raw data from iCloud is always zero so it looks like Apple is not getting it from them. 

```
2023-05-24 02:25:05 [pyicloud_ic3:1359]  ICLOUD > ──────── FAMSHR DEVICE - OFFLINE/NO LOCATION DATA - <GARY-AIRPODS> ────────
2023-05-24 02:25:05 [pyicloud_ic3:1359]  ICLOUD > {'▶ITEMS◀ (items)': {'id': 'AbrWudzuqf...', 'modelDisplayName': 'Accessory', 'lostModeCapable': False, 'name': 'Gary-AirPods', 'deviceClass': 'Accessory', 'deviceStatus': '201', 'rawDeviceModel': 'AirPods_8212', 'batteryLevel': 0.0, 'deviceDisplayName': 'AirPods Pro', 'prsId': None, 'batteryStatus': 'Unknown', 'location': None, 'deviceModel': 'AirPods_8212-0'}}
```

**Comment by pitoganzado** [comment]
![image](https://github.com/gcobb321/icloud3_v3/assets/17479038/71118a58-1797-490b-a028-b18ad0cbf123)
how can i fix this message?
thank you

**Comment by JtwoA** [comment]
Have you done steps noted as 1, 2, and 3?

John Alan
[image]

01000011 01110101 01110010 01101001 01101111 01110011 01101001
01110100 01111001 00100000 01101001 01110011 00100000 01110100
01101000 01100101 00100000 01101101 01101111 01110011 01110100
00100000 01110110 01100001 01101100 01110101 01100001 01100010
01101100 01100101 00100000 01100110 01100101 01100001 01110100
01110101 01110010 01100101 00100000 01101111 01100110 00100000
01101001 01101110 01110100 01100101 01101100 01101100 01101001
01100111 01100101 01101110 01110100 00100000 01101100 01101001
01100110 01100101 00101110

------- Original Message -------

**Comment by gcobb321** [comment]
As John said, your browser needs to be refreshed.

**Comment by amaisano** [comment]
Hi @gcobb321 ! Loving the updated UI/UX for v3.

One issue though - I can't use my old interval of 1 minute for my Apple Watch. 

I have issues when arriving home where it takes 5 minutes for my house to see I'm here, so for 5 minutes I get tons of security alerts "SOMEONE IS BREAKING INTO YOUR HOME" - 1 minute is what I used in v2 and it was perfect -- enough time for me to pull in the driveway and stop the car.

Is is possible to set a lower interval for when I arrive home / am home / leaving home zone?

**Comment by JtwoA** [comment]
Did you not have enormous battery drain setting the Apple Watch interval to 1 minute?

John Alan
[image]

01000011 01110101 01110010 01101001 01101111 01110011 01101001
01110100 01111001 00100000 01101001 01110011 00100000 01110100
01101000 01100101 00100000 01101101 01101111 01110011 01110100
00100000 01110110 01100001 01101100 01110101 01100001 01100010
01101100 01100101 00100000 01100110 01100101 01100001 01110100
01110101 01110010 01100101 00100000 01101111 01100110 00100000
01101001 01101110 01110100 01100101 01101100 01101100 01101001
01100111 01100101 01101110 01110100 00100000 01101100 01101001
01100110 01100101 00101110

------- Original Message -------

**Comment by amaisano** [comment]
I do on my iPhone which is connected to the watch, but the watch itself was fine. made it 2 days.

I go for walks and leave my iphone at home too, which is originally why I wanted to use this component. If i always had my phone with me, i'd just use the Geocoded Location track that comes with the ios app which is faster than the 5 minute min for icloud.

Unless you have a better option - like something that instantly switches zones when i cross over, but otherwise keeps the polling at a reasonable level at home...

**Comment by JtwoA** [comment]
Your use case is exactly mine. I have the cellular Ultra Apple Watch and would love to leave my phone behind for runs/rides.

John Alan
[image]

01000011 01110101 01110010 01101001 01101111 01110011 01101001
01110100 01111001 00100000 01101001 01110011 00100000 01110100
01101000 01100101 00100000 01101101 01101111 01110011 01110100
00100000 01110110 01100001 01101100 01110101 01100001 01100010
01101100 01100101 00100000 01100110 01100101 01100001 01110100
01110101 01110010 01100101 00100000 01101111 01100110 00100000
01101001 01101110 01110100 01100101 01101100 01101100 01101001
01100111 01100101 01101110 01110100 00100000 01101100 01101001
01100110 01100101 00101110

------- Original Message -------

**Comment by gcobb321** [comment]
Here is a preview of beta 18. I'll publish it in a few days if everything looks good and there are no new problems. 

The two main big changes are (1) the new 'HA hack' that fixes the problem created when the HA guys changed the recorder log history module history and (2) the additional startup retry methods when missing famshr location data is detected during startup.

Unzip to the custom_components/icloud3 directory.
[icloud3_v3-beta-18.zip](https://github.com/gcobb321/icloud3_v3/files/11750262/icloud3_v3-beta-18.zip)


Change Log:
1.  Event Log - Fixed a problem where displaying a Nearby Devices message would display the text below the Event Log items window due to a HA Front End change.
2.  iOS App Only - Location requests will be sent to the device when the iOS App is the primary data source just like it is to the iCloud Web Service. This only sends a request to the iOS App. If the device is asleep or the iOS App is running in the background or not loaded, it will not respond and send the location. In this case, another location request will be sent when the next update time is reached.
3.  StatZone - Do not create a Stationary Zone when the current location data is old.
4.  History Recorder - Fixed the problem leading to the '_exclude_e' not being found error while adding iCloud3 HA Recorders sensors to the exclude_entities list.
5.  icloud3_debug_log file - Changed the name to 'icloud3-0.log' since it now contains a lot more info than just debug entries. Three generations of the log file will be kept, indicated by the suffix ('-0'=Today, '-1'=Yesterday, '-2'=Two days ago).
6.  Device Verification during startup - When starting up, device data/location information is used to verify the device. If the iCloud Location Service does not respond with location data for one or more devices during startup, the verification process is now automatically retried several times. If it still fails for one or more devices, a restart is needed message is displayed to try again later. The verified devices will continue to be tracked.
7.  Configuration Wizard - Change the range of the inzone interval parameter values from 5-240 minutes to 1-300 minutes.

**Comment by JtwoA** [comment]
@amaisano - I'd like to know more how you have your apple watch configured.  I bumped mine down to InZone every minute and went for a 12 mi bike ride, coming back in an hour and finding 30% of my battery gone on my Ultra.  I was running a workout tracker but I do that every night and it hasn't had that result.

Curious how you're configured in iC3.

**Comment by gcobb321** [comment]
I found some HACS configuration problems, have fixed them and v3 is now showing up on my system in the Integrations category (along with v2). I had to readd iC3 as a custom repository, download it and restart HA.

See [here](https://gcobb321.github.io/icloud3_v3/#/chapters/2.0-installing-and-configuring?id=step-1-install-icloud3) for information about adding the custom repository.

**Comment by marceloaf** [comment]
Cool!! Tks

**Comment by amaisano** [comment]
Does that mean we can update to newer betas via the UI / and/or get
notifications on new versions now?

**Comment by gcobb321** [comment]
I hope so. I just updated to b18d to test everything. Will probably know tomorrow. I still can not get the iCloud3 description and info to display when iCloud3 is selected. Looking at the .storage/HACS.repository file, the config I submit does not get updated. I’m going to let HACS run overnight without touching anything to see what happens. If it is still not updated, I’ll be in touch with the HACS guys.

**Comment by amaisano** [comment]
Hey it's working! I think! I just got this a short while ago and hit "Install" will post once restarted to see if it works...

![image](https://github.com/gcobb321/icloud3_v3/assets/688435/a39d9ce7-ae9d-4df3-90b6-7f678bd05774)

**Comment by gcobb321** [comment]
The HACS repository file that stores the config may not get updated automatically. If iCloud3 v3 shows as installed but the description (README file) it’s not displayed and all you see at the top is text with the version and my GitHub user I’d (v3.0.0-beta-18c, @gcobb321, ### ,###) you need to force HACS to do an update. 

To update the iCloud3 HACS config go HACS > Integrations > 3-dots upper right > Custom Repository > iCloud3 > 3-dots upper right > Update information

**Comment by gcobb321** [comment]
@Snuffy2
Some files for you with the new recorder_prefilter.

[Places files - recorder_filter & sensor](https://github.com/gcobb321/icloud3_v3/files/11872113/recd_prefilter_sensor.zip)

[icloud3_v3-beta-19.zip](https://github.com/gcobb321/icloud3_v3/files/11872120/icloud3_v3-beta-19.zip)

**Comment by Snuffy2** [comment]
This looks great and, on initial testing, seems to work well. Thanks!

**Comment by Snuffy2** [comment]
Perfect! I'm going to add it as a git submodule to places.

**Comment by amaisano** [comment]
@JtwoA the 5 minute interval on my watch allowed me to use my watch for about 24 hours, full to empty.

However I noticed this past week since I was able to change the interval to 1 minute, it's been reduced to more like 16 hours.

AND STILL i'm getting reported as an intruder by my home automations when I return home, because 1 minute is still not accurate enough to detect me as home between parking the car and opening the doors.

Here's the issue:

![image](https://github.com/gcobb321/icloud3_v3/assets/688435/4c393a71-e499-4b49-9671-d1283866a571)

You can see the storm door is opened between being "Away" and "Home" on the iCloud3 watch tracker. The iOS companion app is the fastest and is basically instantaneously showing me as home before I even get out of the parked car.

@gcobb321 is there something I can configure to make the -to-home-zone transition snappier?

If not I'll likely create a node-red sensor that uses my iOS companion location if certain criteria is met. 75% of the time I have both my iPhone and Watch on me, so the companion app is good.

It's the 25% of the time I go out for walks and runs WITHOUT my phone that i need/use the iCloud3 tracker for.

**Comment by JtwoA** [comment]
I've been testing this and have been able to run the full day plus a bit with my Ultra (running beta WatchOS 10).

- I'm set to one minute.
- When I return from walks it takes up to two minutes for my Watch to be reflected at home at which point my house switches over to HOME, alarm disarmed, etc. (The alarm people have been very patient with frequent misfires last week.)

I, too, carry the phone probably 90% of the time. It's bike rides locally and quick trips to workout that it's Watch-only.

------- Original Message -------

**Comment by gcobb321** [comment]
@amaisano 

There are some sample automations you might want to look at in the iCloud3 docs. The one I use for opening my garage door checks if I’m entering the Home zone OR if the home distance sensors is less than .2. There is also a check to make sure I was driving and had gone 3 more than miles from home.  You will never get a fast enough response with just a zone enter because of delays responding to internet location requests, iosapp polling delays, gps accuracy, etc. When you get within 1 mile from home (it may be less), iCloud3 goes to a 15-secs polling interval so you don’t really need a 1-min interval all the time. Follow the events in the Event Log to see what is going on.

**Comment by JtwoA** [comment]
Gary, which of these are you referring to by "home distance sensor is less than .2"?

[image.png]

------- Original Message -------

**Comment by gcobb321** [comment]
I use the `gary_iphone_home_distance` sensor. See [here](https://gcobb321.github.io/icloud3_v3/#/chapters/5.0-sample-automation-scripts?id=gary-arrives-home-automation-homedistance-sensor) for my Automation to open my garage door.

**Comment by DJBenson** [comment]
I've installed the latest beta from HACS and am getting the below error for all my devices;
<img width="393" alt="image" src="https://github.com/gcobb321/icloud3_v3/assets/1013909/ec215cac-5cc8-4d19-8c29-310d8cb8a704">

**Comment by gcobb321** [comment]
Check the _icloud3-0.log_ and _home-assistant.log_ files for other error messages before that one. There must have been some others during startup up. Turn on debug logging in Event Log > Actions > and restart HA. If problem continues, email the _icloud3-0.log_ file to geekstergary@gmail.com.

**Comment by DJBenson** [comment]
The icloud3.log file is spammed with the same error message. In the home-assistant.log file I see this;

`(MainThread) [homeassistant.components.device_tracker] Platform icloud3 does not generate unique IDs. ID icloud3_myphone is already used by device_tracker.myphone - ignoring device_tracker.myphone`

For context I migrated from v2 to v3 so not sure if this is related.

**Comment by JtwoA** [comment]
@gcobb321 why does my Watch, set to 1 minute, take 3 minutes when I'm arriving home to discover I'm arriving home?  My phone was with me and picked right up on it.  

I'm using home_device_sensor on both set to .1MI.

![image](https://github.com/gcobb321/icloud3_v3/assets/12992040/cfe8e5a0-94b8-4489-82cf-dad66258c6f5)

**Comment by gcobb321** [comment]
Because your direction of travel was Away. If distance is < 2km and going Away, the interval is set to the ‘old location thresholds’ value in your config (3 min). The ‘Method’ field in the Event Log Results shows how the interval is calculated. It’s been that way for a long time. 

Away is set by comparing the distance of this update with the previous one. If greater than Away. If less than Towards. That is based on the gps from your iCloud account. Maybe previous was old or had poor gps accuracy.

**Comment by JtwoA** [comment]
Interesting as I had been gone from Home and in another zone completely for over half an hour.

>

**Comment by smarthomefamilyverrips** [comment]
Hello @gcobb321 

I did install the integration manualy using the b19.2 version but it not shows me the battery percentage of my Iphone 14 PRO .... it stays on 0% ..... any idea how to solve this? Thanks upfront!

**Comment by gcobb321** [comment]
@smarthomefamilyverrips 
1. The iosapp battery sensor I is checked every 5-secs. The sensor that is checked is shown in Stage 5 of the startup process.  Make sure that is the correct sensor entity and it’s value is changing. 
2. It is also updated from the FamShr data when the device’s location is updated. It is displayed the Results line. If it is correct there but not correct on the sensor.[devicename]_battery state value on the Lovelace dashboard, the wrong sensor name is probably being used. Check the Developer Tools > States screen and Devices z& services > Entities screen for an incorrect or duplicate entity name. 
3. Check the Attributes of the battery sensor on the Lovelace Dashboard screen. There is some info on when it was last updated. 
4. Check the HA Log file for any errors related to the battery.

**Comment by smarthomefamilyverrips** [comment]
@gcobb321 

> @smarthomefamilyverrips
> 
> 1. The iosapp battery sensor I is checked every 5-secs. The sensor that is checked is shown in Stage 5 of the startup process.  Make sure that is the correct sensor entity and it’s value is changing.

You are referring to the HA IOS app?

> 2. It is also updated from the FamShr data when the device’s location is updated. It is displayed the Results line. If it is correct there but not correct on the sensor.[devicename]_battery state value on the Lovelace dashboard, the wrong sensor name is probably being used. Check the Developer Tools > States screen and Devices z& services > Entities screen for an incorrect or duplicate entity name.

I only find one battery entity with unique name giving value 0

> 3. Check the Attributes of the battery sensor on the Lovelace Dashboard screen. There is some info on when it was last updated.

Seems like not get updated at all

data_source: iCloud3
sensor_updated: '2023-07-15 12:52:31'
battery_status: ''
battery_data_source: ''
battery_level_updated: '00:00:00'
device_class: battery
state_class: battery
unit_of_measurement: '%'
friendly_name: Lory Iphone New Battery

> 4. Check the HA Log file for any errors related to the battery.

No errors related to battery nor integration in logs

**Comment by gcobb321** [comment]
The sensor for the ios battery App that is monitored is shown in Stage 5. An example is in the iCloud3 docs [here](https://gcobb321.github.io/icloud3_v3/#/chapters/1.4-evlog-during-startup?id=stage-5-configure-tracked-devices). 

At the bottom of the screen Image for _Tracked Device: gary_iphone (Gary/iPhone)_ shows that the _Battery: gary_iphone_app_battery_level_  . That is the iOS app sensor that is being monitored and it is created and updated by the iOS app. Verify that that sensor exists and is being updated. If not, refer to the iOS App docs regarding that. 

Also look on the Event Log screen to see if the Battery level shows there. It’s value in listed on the _Results_ line below the red line that shows the zone, travel time and distance when a device is tracked.

**Comment by smarthomefamilyverrips** [comment]
@gcobb321 

I have a entity battery level that is reported by the HA app but this not updates regularly now is on 64% while phone is on  62%

I now did add the device also at Family Share field and now get a value at battery, it only not updates the value... it stays on 63% while phone already jumped to 62%

Also where I find event log screen?

**Comment by gcobb321** [comment]
@smarthomefamilyverrips 
The iOS App battery_level sensor is monitored every minute, the FamShr is only updated when the device is located, 

You can find info on the Event Log [here](https://gcobb321.github.io/icloud3_v3/#/chapters/1.1-ic3-components?id=event-log) and [here](https://gcobb321.github.io/icloud3_v3/#/chapters/2.0-installing-and-configuring?id=step-53-create-the-event-log-portion-of-the-lovelace-screen) and [here](https://gcobb321.github.io/icloud3_v3/#/chapters/1.4-evlog-during-startup?id=event-log-during-startup)

**Comment by smarthomefamilyverrips** [comment]
@gcobb321 

After some trying and tweaking HA app it seems all is working as expected.... Thank you for your help!

**Comment by gcobb321** [comment]
I'm glad you got it working. Also, I'm working on a FAQ for the iCloud3 docs and would like to include what you did to get it working. Then, next time someone runs into the problem, I could just point them to that answer.

Thanks.

**Comment by smarthomefamilyverrips** [comment]
@gcobb321 

The problem with not having any value for battery ( 0%) was solved with choosing the iPhone as device in FamShare field in device config of integration

To have more frequently updates I did change the update interval when present in a zone to from the standard 120 minutes to 5 minutes. 

Also I used the "IOS ShortCuts App"  to set some automations to trigger sensor updates of IOS HA App (ex. when open Facebook App then Update HA Sensors) 

These settings and tweaks made it work for my use case of the iCloud3 V3 integration.

**Comment by strickersuk** [comment]
I have one device not showing up in the Event Log, my wife's iPhone. The device is showing and the logbook is updating, her watch and iPad are working. Any ideas? Great work by the way, looking forward to when it becomes the default iCloud option.

**Comment by strickersuk** [comment]
Is there a maximum of 10 items at the top of the event log?

**Comment by gcobb321** [comment]
@strickersuk 
Yes. The max is 10 devices for the Event Log and Configurate Settings > Update Devices screen

**Comment by smarthomefamilyverrips** [comment]
@gcobb321 if I first did manual copy the folder in to custom folder can I then now use repository in HACS to update to latest release? Or should I again copy the folder manually?

**Comment by gcobb321** [comment]
You can if you set iCloud3 up as a custom repository. Make sure the one you select says **iCloud3 v3**, **not iCloud3 v2** when you install from HACS.

**Comment by smarthomefamilyverrips** [comment]
@gcobb321 why the integration add this file in /config "known_devices.IC3.CAN.DELETE.yaml" and deletes big part of devices from original "known_devices.yaml" file? ..... 2nd question can the newly added file be deleted?

**Comment by gcobb321** [comment]
@smarthomefamilyverrips  the known_devices.yaml file is a leftover file from older versions of HA before integrations and probably isn’t being used anymore. The current versions of HA still read it and it will cause problems creating the iCloud3 v3 device_tracker entities.  HA thinks the iCloud3 devices names still exist when they really don’t and add a suffix to it that breaks automations and scripts

Example:
if HA has added the iCloud3 v2 gary_iphone device to known_devices, HA would think gary_iphone still existed and create gary_iphone_2 when iCloud3 started. iCloud3 would update the _2 just fine but the automations and scripts are still looking for the gary_iphone and no longer work. 

If everything in HA works fine and all your other integrations are ok, you can delete it.

**Comment by smarthomefamilyverrips** [comment]
@gcobb321 I assume we both talk about deleting the "known_devices.IC3.CAN.DELETE.yaml" file created when i did install iCloud3 V3 (FYI I never used V2 and installed directly V3) and not to delete the original known_devices.yaml ..... if I would delete the original I will lose all device trackers available in my HA install .... so just to be clear my question is what to do with the by the integration added "known_devices.IC3.CAN.DELETE.yaml"

**Comment by gcobb321** [comment]
Is there anything in the file? If it empty, delete it. If it is not empty and you have a known_devices.yaml file, are they the same? If yes, delete it. If they are different, is the new k_d file newer and the same date as the iCloud3 install? If it it newer, it was recreated by HA and delete it after running HA and checking your other device trackers. 

As far as I know, that file is a legacy file that HA uses to create entity names.

**Comment by smarthomefamilyverrips** [comment]
> Is there anything in the file? If it empty, delete it. If it is not empty and you have a known_devices.yaml file, are they the same? If yes, delete it. If they are different, is the new k_d file newer and the same date as the iCloud3 install? If it it newer, it was recreated by HA and delete it after running HA and checking your other device trackers.
> 
> As far as I know, that file is a legacy file that HA uses to create entity names.

@gcobb321 I did make a copy of the "known_devices.IC3.CAN.DELETE.yaml" to my PC to be sure, after deleted the file from /config folder and restarted HA.... all started up normal and all my device_trackers are still available, so I think is safe to conclude that the "known_devices.IC3.CAN.DELETE.yaml" can be deleted from /config folder without any problems.... Thanks for your help!

**Comment by Seger85** [comment]
![image](https://github.com/gcobb321/icloud3_v3/assets/35084198/50fddb21-4fef-42fe-bc71-9a0f969bd3e4)

Good morning,

first of all, thank you very much for your breathtaking work for the community, just great! I am currently using the latest version v3 and I noticed that it changes the state of charge (akku) with a small offset first. I have set it to update every minute and I am at home. The status changes as shown in the picture but not immediately after the update but about 1min. later. I think it used to be that this status changed as soon as I connected a charging cable, regardless of the update interval. Do you have a tip or an explanation for me?

Best regards
Seger

**Comment by Seger85** [comment]
just see an explanation above, but was it like that before, had it in real time in my memory. But maybe it's too exaggerated.

**Comment by Seger85** [comment]
Okay super, ich werde es heute Abend testen und danke, dass du dir diesen Teil nochmal angeschaut hast.
    	
    
  

 Von: Gary Cobb ***@***.***>Gesendet: Montag, August 21, 2023 21:49An: gcobb321/icloud3_v3 ***@***.***>Cc: Seger85 ***@***.***>; Mention ***@***.***>Betreff: Re: [gcobb321/icloud3_v3] iCloud3 v3 - General Questions & Help With Problems (Issue #129) 
@Seger85 iCloud3 does monitor the iOS App battery level and status (charging/not charging) looking for changes every 5-secs. It should be posting a message to the Event Log when the charging/not charging changes or when the battery level ends in 0 or 5.
There were several problems that have been fixed. (1) The charging/not charging was being identified as changed but then not used to generate the Event Log message and (2) If the phone was plugged in, then unplugged, then plugged in again, the 2nd plug in msg was not being displayed.
These have been fixed. Go to the v3 Development Version Releases page here and download the patch-pr1.2-battery-msg.zip file. Unzip it into the icloud3 directory and restart HA.
I have changed the way iC3 determines if the device is charging or not. Previously, the iosapp_battery_status  sensor entity was monitored. When ic3 gets the iOS App battery level sensor entity, it also gets the icon. The icon has 'charging' in it. iC3 now uses that to determine if is charging or not. Let me know if there are any issues with this method and I'll go back to the old way.

**Comment by Seger85** [comment]
Okay great, I'll test it tonight and thanks for looking at that part again.
    	
    
  

 Von: ***@***.*** ***@***.***>Gesendet: Dienstag, August 22, 2023 07:51An: gcobb321/icloud3_v3 ***@***.***>; gcobb321/icloud3_v3 ***@***.***>Cc: Mention ***@***.***>Betreff: Re: [gcobb321/icloud3_v3] iCloud3 v3 - General Questions & Help With Problems (Issue #129) 
  
  
  
    
    	
    	Okay super, ich werde es heute Abend testen und danke, dass du dir diesen Teil nochmal angeschaut hast.
    	
    
  

 Von: Gary Cobb ***@***.***>Gesendet: Montag, August 21, 2023 21:49An: gcobb321/icloud3_v3 ***@***.***>Cc: Seger85 ***@***.***>; Mention ***@***.***>Betreff: Re: [gcobb321/icloud3_v3] iCloud3 v3 - General Questions & Help With Problems (Issue #129) 
@Seger85 iCloud3 does monitor the iOS App battery level and status (charging/not charging) looking for changes every 5-secs. It should be posting a message to the Event Log when the charging/not charging changes or when the battery level ends in 0 or 5.
There were several problems that have been fixed. (1) The charging/not charging was being identified as changed but then not used to generate the Event Log message and (2) If the phone was plugged in, then unplugged, then plugged in again, the 2nd plug in msg was not being displayed.
These have been fixed. Go to the v3 Development Version Releases page here and download the patch-pr1.2-battery-msg.zip file. Unzip it into the icloud3 directory and restart HA.
I have changed the way iC3 determines if the device is charging or not. Previously, the iosapp_battery_status  sensor entity was monitored. When ic3 gets the iOS App battery level sensor entity, it also gets the icon. The icon has 'charging' in it. iC3 now uses that to determine if is charging or not. Let me know if there are any issues with this method and I'll go back to the old way.

**Comment by smarthomefamilyverrips** [comment]
> @Seger85 iCloud3 does monitor the iOS App battery level and status (charging/not charging) looking for changes every 5-secs. It should be posting a message to the Event Log when the charging/not charging changes or when the battery level ends in 0 or 5.
> 
> There were several problems that have been fixed. (1) The charging/not charging was being identified as changed but then not used to generate the Event Log message and (2) If the phone was plugged in, then unplugged, then plugged in again, the 2nd plug in msg was not being displayed.
> 
> These have been fixed. Go to the v3 Development Version Releases page [here](https://github.com/gcobb321/icloud3_v3/releases) and download the `patch-pr1.2-battery-msg.zip` file. Unzip it into the icloud3 directory and restart HA.
> 
> I have changed the way iC3 determines if the device is charging or not. Previously, the iosapp_battery_status sensor entity was monitored. When ic3 gets the iOS App battery level sensor entity, it also gets the icon. The icon has 'charging' in it. iC3 now uses that to determine if is charging or not. Let me know if there are any issues with this method and I'll go back to the old way.

@gcobb321 should we all apply the patch you mentioned here?

**Comment by gcobb321** [comment]
Yes, You can apply it.

**Comment by Seger85** [comment]
hey everyone, quick feedback. After the patch, I noticed the following: although it updates in the log every minute, unfortunately the new value, which is higher in this case, is not set. Unfortunately, the ios app does not update at all, at least not within a minute. The update only takes place after the charging cable has been disconnected. unfortunately, the charing mode is not taken over as a value from the log. enclosed are some photos:

![Unbenannt](https://github.com/gcobb321/icloud3_v3/assets/35084198/403bb670-c82c-4297-9ae3-ef53a47ac422)
![image](https://github.com/gcobb321/icloud3_v3/assets/35084198/48d3ec0e-3e0e-4dbd-84cf-bbb6eaa9cd4a)

**Comment by gcobb321** [comment]
@Seger85 The battery level only displays in the Event Log when it ends in 0 or 5 or the charging/not charging changes. If it did display a message on every change, the log would fill up with battery level messages. The battery sensor is updated on every change, as your entity shows,  and the value is displayed in the tracking results message. 

On another note, I noticed that every item in the Event Log starts with >home on the left side. That is where the time displays for items not related to displaying the tracking info (inside the blue box area. See the screenshots in the  iCloud3 User Guide. **Click (?)** Icon. I would like you to send me to your _icloud3-0.log_ file to geekstergary@gmail.com so I can see what is going on.  Also, do an Event Log > Actions > Export Event Log and send me that file too.

**Comment by Seger85** [comment]
I don't quite understand it, so a new number is recorded on the right-hand side, but not on the left-hand side (in the widget). So the log works, but the value of the entity doesn't change. regarding the update of the ha app, i just found a reason. i still had an old local ssid of my wlan set in the app, so there was no local push of the app and an update doesn't really take place via the external connection, i guess. So the app and also the update of the standard app works now. i'll send you all my logs right away... THANKS

**Comment by gcobb321** [comment]
@Seger85 
I have fixed the Event Log issue and posted a new patch (_patch.pr1.3-08.23+event-log.zip_) on the GitHub releases page (here)[https://github.com/gcobb321/icloud3_v3/releases/tag/v3.0-pr1.1]

Download and unzip into the icloud3 directory. A browser refresh will be requested.

Note: _patch.pr1.3-08.23-xxx_ includes the previous _patch.pr1.2-08.21-xxx_)

**Change Log:**
1. Battery - Fixed a problem where the battery level/status was not being updated immediately in the Event Log when it went from 'charging' to 'not charging' or 'not charging' to 'charging'.
2. GPS - The 'Show GPS Coordinates' parameter is now applied when the event is displayed in the Event Log instead of when it is added to the Event Log. You can disable this parameter so they are not normally displayed. Later, if you need to review where a device has been, reenable the parameter and the location coordinates for previous events will be displayed.
3. Event Log - Fixed a problem where the event time was not displayed if the 'Zone Display-as' parameter was set to display the zone entity_id.
4. Event Log - Fixed a problem with the Event Log Card not detecting a version update.

**Comment by gcobb321** [comment]
### I have just uploaded Prerelease v1.3

This is an update to _pr1.1_. It includes the _pr1.2_ and _pr1.3_ patch files.

**Comment by gcobb321** [comment]
### I have just uploaded Prerelease v1.3

This is an update to _pr1.1_. It includes the _pr1.2_ and _pr1.3_ patch files.

**Comment by Seger85** [comment]
Good morning everyone, I would like to give you feedback that I think it is running much better than before, especially the transfer of the value from the log as soon as the iOS app reports something works better on the home WLAN, great! I'll be happy to send you all my logs again this evening, you might find some clues about outliers there.

I also have two more topics/questions:

The value "MovedDistance" on the IPhone tracks the current metres for me, but shouldn't it actually represent the sum of all metres for a day? Or should I build it myself using a helper?

I also noticed that the waze distance to the Work zone was also listed as an entity at the time. Does this function no longer exist, or have I not ticked off a point?

Thank you and best regards
Seger

**Comment by gcobb321** [comment]
1. No need for the logs. iCloud3 reports the data when it changes. The iOS app is not in constant communication when not home and on a Logan. 
2. Moved distance is the distance since the last locate and used to try to determine the direction, not to keep up with Joe far you traveled. You will have to do that yourself if that is important. 
3. See the Configure Settings > Sensors to turn on the Waze sensor, and others.

**Comment by gcobb321** [comment]
@knozturk 
You can rename the stationary zone on the Configure Settings > Page2 > Special Zones screen. 

But, the Stationary zone is used to create a zone for a short time, not for a regular place you normally spend a lot of time at. Examples are a doctors office, friends house, a store, etc. You should probably set up an HA zone for places like work, school, parents house, club, gym, etc

**Comment by RafhaanShah** [comment]
Is it possible to receive the 2FA code via SMS rather than on a device?

**Comment by thelegendk** [comment]
Thanks for the update! 
I am getting an error when trying to add device.
Any idea why it could be? I have 2FA, companion app installed, location permissions are good and have 4 people total in iCloud family. Log file doesn't show any errors, last comments are Config > Updated FamShr data > RequestedBy-None, then it shows my devices along with *No location for some (non-shared) and *Added for rest. Last line is CONFIG > icloud_account (save) > User input-{'data_source_icloud': ['famshr'], 'username': 'de... etc. 
<img width="421" alt="Screenshot 2023-10-08 235359" src="https://github.com/gcobb321/icloud3_v3/assets/119612823/12c9d2e9-2659-4e6f-9f43-077e2a6252d4">

**Comment by gcobb321** [comment]
That has been fixed. 

Below is v3.0.rc6 which I will be releasing later today or tomorrow. It should be on HACS in a few days when HACS does it's update.

Unzip it into the _config/custom_components/icloud3_ directory and restart HA.

[icloud3_v3.0.rc6.zip](https://github.com/gcobb321/icloud3_v3/files/12846884/icloud3_v3.0.rc6.zip)

**Comment by tokroth** [comment]
Hi!

I am new to icloud3. Spent some time setting it up with various issues. I am using v3 pr 1.4. I have a few questions/issues that I cannot find answers to above here or in the docs. Maybe there are a some answers here in the forum?

1. 2 times in the past 3 days my daughter's phone stops receiving location through FamShr id. All other family members continue to work normally. iCloud3 starts polling apple servers aggressively - once every 5 seconds and after a few hours the famshr dev id in the iCloud3 config gets removed and her famshr location is never updated again until I go in manually and reconfig her device in the UI? My hope is that this is a bug that will get resolved...

2. Given number 1 above (but also completely normal use cases like a) my kids set devices in power saving when they are on the way to school since they cannot use them during the day or b) lost cell coverage in buildings / tunnels etc.): I quite often have the icloud3 device tracker with a wrong state just because that was the last known location for the famshare device. Even if I have set up a HA companion app that has a MORE recent location update, the device tracker still shows an older last known location from the famshr-device? Shouldn't the device_tracker ALWAYS have the zone state from the last successful GPS poll (IOSAPP or FamShr etc.)? Otherwise I do not understand the point of setting up the IOSAPP device at all.. What am I missing?

2. What is the point of making the device_tracker entity in apple get a state update when the location is NOT updated? As a work-around for my problems above I rely on multiple device_trackers (so that I at least have "home"/"not home" correct) for each person. But icloud3 keeps updating its device_tracker even though the location is not new? Unfortunately the person entity in HA will be updated with the last updated device_tracker regardless if the attribute "last located" in the icloud3 device_tracker is many hours ago... 

4. iCloud3 seems unable to get a FnF list even though it is set up on my phone. Anyone knows why? I can only select FamShare when configuring devices?

**Comment by gcobb321** [comment]
1. When iCloud can not locate a device, it keeps sending the last known location. iC3 sees it is old and goes into an error cycle starting at 5-sed, then 30, then 1-min, 5-min, 30, 1-hour, etc. After a while, it starts over and continues. I have not seen where the device info is changed or deleted from the configuration when this happens.. iC3 uses the data it receives from Apple and the iOS App. When the data is old or there is no response, there is not much iC3 can do.

Are there any errors in the home-assistant.log or icloud3-0.log file? Turn on Log Level-Debug (_Configure Settings > Format Parameters_) and when you think the device info is deleted again, email the _icloud3-0.log_ file to _geekstergary@gmail.com_ and I'll take a look at it.

2. The iC3 state (Home, Away or a zone) is set 2 ways depending on the parameter on the _Configure Settings Format Parameters_ screen. HA will set it based on the gps or iC3 will set it using the last zone iC3 says the device is in. When your devices go into airplane mode, lose there signal (power saving, tunnel), the iC3 zone sensor (and dev_trkr based on the selection) will not change from one value to another. The last one is retained. When you say it changes, what entity are you looking at? Did it change in the HA history? Did the sensor.[devicename]_zone also change? 
3. Again, the device_tracker state value can be controlled by HA, not iC3. The Person entity is controlled by HA, not iC3. When iC3 updates it's device_tracker entity, a zone/device trigger is initiated by HA. HA looks at the gps and determines if it is in a zone or not. It does not look at the age or gps accuracy. iC3 does look at the gps accuracy and will discard a location if it is old. Could the person be updated by another device_tracker?  

Regarding the school. Can you make a school zone that is big enough so your son's phone will go into that zone when it is turned off.

**Comment by tokroth** [comment]
I just saw in my logs that when I restarted HA IC3 seems to be in  pr 1.3. I thought I was on 1.4 since 2 days ago...?

2023-10-09 22:09:42 [start_ic3_co:0354]  INITLOC > Requesting Initial Locate
2023-10-09 22:09:42 [start_ic3_co:0356]  INITLOC > Initializing iCloud3 v3.0pr1.3 > Complete

So I re-downloaded 1.4 from HACS - and Now I got this update board in the HA settings - see below. Very strange. Here it says that I am on 3.0 pr1.4 but latest is 2.4.7???

Which version is correct? What should I do?

iCloud3 Device Tracker update
2 hours ago
v2.4.7
Installed version
v3.0-pr1.4
Latest version
v2.4.7
[Read release announcement](https://github.com/gcobb321/icloud3/releases/v2.4.7)
Notes:

This is an update to Prerelease Version pr1.3 that fixes a few bugs and changes the way Stationary Zones are handled.

Change Log:

Bug Fix - Fixes a "type object 'GlobalVariables' has no attribute 'this_update_tine'" error. it should be '_time', not '_tine'

**Comment by gcobb321** [comment]
v2.4.7 is the latest actual release. pr1.4 is set as a Prelease . You see that because you have beta versions enabled. FYI- I uploaded v3.0.rc6 a few hours ago which might show up in HACS tomorrow after it refreshed everything.

**Comment by SimoHDK** [comment]
Has anybody else seen issues with 'Centre in Zone' not functioning? 

Despite being turned on, I'm seeing the scatter effect. Devices are staying in 'Home' zone just boucning around as GPS adjusts.

<img width="1167" alt="Centre in Zone On" src="https://github.com/gcobb321/icloud3_v3/assets/17811281/308c8422-cabc-46bc-a62d-0e194e2a2c34">

**Comment by JtwoA** [comment]
I had a rare occurrence two nights ago wherein my house shut down because I was determined to be not home. That hasn't happened in many, MANY months.

John Alan

Sent with [Proton Mail](https://pr.tn/ref/TYCHKJZF3G2G) secure email.

------- Original Message -------

**Comment by JeanMertz** [comment]
Has anyone had any issues with the battery sensor triggering an update event every 5 seconds?

Comparing the two events separated by 5 seconds, the only attribute that has changed is `sensor_updated`, (as expected, the last one is 5 seconds later than the first one).

Is there any way to either

1. reduce the polling/update frequency of this event, or
2. ignore `sensor_updated` when determining whether an update event should be triggered in HA?

I should note I'm seeing this as I have a system subscribed to all state changes through the HA Websocket API.

The frequent change events happen for the following two entities:

- `sensor.xxx_iphone_icloud_battery`
- `sensor.xxx_iphone_icloud_battery_status`

It happens for multiple devices, all have the HA app installed as well.

**Comment by amaisano** [comment]
Noticing a recurring issue that maybe someone else here can help with:

When my Apple Watch dies (tracked as `adamo_watch`), then I charge it and put it back on my wrist and unlock it, iCloud3 stops all tracking on it. I usually charge overnight after the watch dies, but it seems (to me, so far), that if I let the watch die completely, iCloud3 won't track it (no exit or enter events, always stays in the pre-death location). Not until I do a restart of iCloud3 at least.

What do you think? This wasn't a problem prior to my iPhone 13 mini -> iPhone 14 mini update, which I just performed last week.

**Comment by gcobb321** [comment]
The iPhone might have the iOS app, which would send a location update where the watch does not. That would cause the phone to relocate when it waddles up. 

Try this. Create a locate automation for the watch that fires at a specific time to try to wake up the watch. See the Service Call section of the iCloud3 docs for some samples.

**Comment by amaisano** [comment]
> @amaisano
> 
> There are some sample automations you might want to look at in the iCloud3 docs. The one I use for opening my garage door checks if I’m entering the Home zone OR if the home distance sensors is less than .2. There is also a check to make sure I was driving and had gone 3 more than miles from home. You will never get a fast enough response with just a zone enter because of delays responding to internet location requests, iosapp polling delays, gps accuracy, etc. When you get within 1 mile from home (it may be less), iCloud3 goes to a 15-secs polling interval so you don’t really need a 1-min interval all the time. Follow the events in the Event Log to see what is going on.

FWIW, (and I'm still experimenting with your suggestions on less than .2, etc), here are the offsets for device responsiveness when arriving home:

![image](https://github.com/gcobb321/icloud3_v3/assets/688435/6aa407cd-4804-43ea-8371-8747efa16c0a)

1. IOS companion app Geocoded Location is the first to change, at `12:51:56`
2. Next is the iCloud3 device tracker for the same phone, at `12:52:00`
3. Last is the iCloud3 device tracker for the Apple Watch, at `12:53:05`

> within 1 mile from home (it may be less), iCloud3 goes to a 15-secs polling interval

I guess I'm not seeing that anecdotally. It took over a minute for that update, but maybe it's because I only traveled less than 1 mile to get coffee and came back.

**Comment by amaisano** [comment]
What does ""icloud3" "icloud location tracking is not available" mean? This is what happens when I try to run that service you suggested

**Comment by gcobb321** [comment]
@amaisano 
That is displayed when:

- The device you are using it’s an automation  does not have the iCloud FamShr device parameter set up in the configuration
- The iCloud username and/or password is not set up so the iCloud is not being used as a data source.

**Comment by amaisano** [comment]
@gcobb321 

It's the former, and the logs are only showing this:

> 2:00:00a	   Service Action > Action-Locate, Options-device(s) using icloud famshr, Device-adamo_watch
> 2:00:00a	   Location request canceled. Old Location Retry is handling Location Updates
> 2:00:00a	   iCloud Location Tracking is not available

This happens after the battery dies on the Watch, then I charge it, put it back on my wrist, and unlock it. It is triggered every time I call (either manually or in an automation) this service call:

<img width="1042" alt="image" src="https://github.com/gcobb321/icloud3_v3/assets/688435/859c395a-a929-4cd2-a894-b11ea3dd9a3b">

The only way I have found so far for my watch to get current location back in order is to completely restart iCloud3v3 :(

**Comment by gcobb321** [comment]
@amaisano 
I see what you mean. My thinking was issuing a `action locate` made no sense in a device was not responding to a lot of previous requests and had an interval timer already set and in an error recovery state. 

The `action resume` does reset everything and clears all error conditions. Try that instead of the ` locate `. I’ve added that to the message to make it clearer that the other option is available. The other option is to do the resume when the device’s error locate interval is above 10-minutes. I’ll give that some thought.

**Comment by amaisano** [comment]
Resume works! Still wondering however if there is a way to detect the "back from the dead" state on the Apple Watch rather than have it polling every 15 minutes for a Resume. Thank you though!

**Comment by JeanMertz** [comment]
> Has anyone had any issues with the battery sensor triggering an update event every 5 seconds?
> 
> Comparing the two events separated by 5 seconds, the only attribute that has changed is the `sensor_updated` timestamp attribute, (as expected, the last one is 5 seconds later than the first one).
> 
> Is there any way to either
> 
> 1. reduce the polling/update frequency of this entity, or
> 2. ignore `sensor_updated` when determining whether an update event should be triggered in HA?
> 
> I should note I'm seeing this as I have a system subscribed to all state changes through the HA Websocket API.
> 
> The frequent change events happen for the following two entities:
> 
> * `sensor.xxx_iphone_icloud_battery`
> * `sensor.xxx_iphone_icloud_battery_status`
> 
> It happens for multiple devices, all have the HA app installed as well.

@gcobb321 any thoughts on this issue?

**Comment by gcobb321** [comment]
@JeanMertz 
It is suppose to only update the battery sensor when the level or status changes. It may also update it with the other device sensors are being updated. I will add some traces and check out how often it is updated.

**Comment by amaisano** [comment]
@gcobb321 the automation that runs Resume on my Watch is not working. The battery ran out last night, and I charged it. I put it back on 3 hours ago, and the script is running every 15 minutes with no conditions that would stop it. The log shows the correct battery, but the HA entities are not being updated:

![image](https://github.com/gcobb321/icloud3_v3/assets/688435/38a4cc6a-ca64-4219-b6ee-0c9dedb87035)

**Comment by gcobb321** [comment]
Actually the location was updating after the resume but iCloud Web Service kept returning an old location. I seem to recall that if a Watch is paired with an iPhone and within a short distance, it tries to get some location info from the paired device. You might also do a locate on the paired device and see if the location becomes current.

I have made a change so when you do a resume, all the location error counts and control values are reset. That will be in rc9 which I have not released yet.

**Comment by gcobb321** [comment]
@amaisano @JeanMertz 
I've wrapped up rc9. Here is a preliminary copy with the code that addresses the problem you had, along with a bunch of other things.  I have not uploaded this to GitHub yet but probably will this coming Wednesday. Let me know how things go (ok or there is a problem).

Unzip icloud3-rc9.zip into the icloud3 directory and restart HA.
[icloud3_v3-rc9.zip](https://github.com/gcobb321/icloud3/files/13467440/icloud3_v3-rc9.zip)

### rc9 Change Log:
1. Stationary Zone
    - Fixed a bug restarting iCloud3 when reinitializing the zones and a Stationary Zone already existed. Some of the zones were not reloaded and the Stage 2 Zone setup process would end.
    - All Stationary Zone information persists over iCloud3 restarts and devices that were in the Stationary Zones will be moved back into it.
    - Unused Stationary Zones will be  checked and deleted every hour
    - A nearby device results will not be used if the device's stationary zone timer has expired
2. The iOSApp Location Time was not being saved correctly when it processed an update handling the Initial Location  triggers when starting iCloud3.
3. Bug fix - Devices that were paired with a Watch that was not being tracked were displaying a message in the Event Log that they were paired with themselves.
4. Battery - Fixed a problem so the battery sensor would only be updated when the battery level or status changed. It was updating when the information hadn't changed which was causing excessive and unnecessary HA battery sensor state changes.
5. Near zone (home) old location update retries - When approaching Home, the location request time was small (15-secs) and old location info was being returned from iCloud, the old location threshold was set to 15-secs to try to get a good location.  This sometimes resulted in the location always being old even when it was good enough to be used. Now, the small interval will only be used for 4 Location requests.  The old location threshold of 2-3 minutes will  then be used.
6. Waze History Database - Removed some maintenance activities being done when iCloud3 starts that was delaying the startup process by almost 2-minutes for a database with 10,000+ entries. This is now done at midnight.
7: Event Log - The Event Log items are displayed a lot faster after selecting a different device or refreshing the screen.
8. HA Area - iCloud3 was assigning or reassigning the tracked and monitored devices to the 'Personal Device' area when starting. This will now only be done when the devices is added or changed from Inactive to Tracked. You can now change or delete the device's HA Area and it will not be reassigned.

9. NEW FEATURE - Away Time Zone - "The time displayed in the Event Log and Sensors show the time an event took place using the Home 'time zone' from your Home Assistant computer.  When you are away from Home and in another time zone, your tracking events are still based on the time at your Home 'time zone', not time in your current location. This feature lets you display time events on the Event Log, Sensors, Sensor Attributes and the Device_Tracker Attribures using your Away location's time zone.

This is useful if you are away for a holiday or business trip to a location in another time zone. For example, you may create a temporary zone for a house you rented, set that zone up as a 'track from' zone on the device's update screen and then all time based tracking events (arrival time, location time, etc) will show your local time instead of your Home time zone time.

The Away Time Zone screen has been added to the Configure Settings, Menu Page 1. On this screen, you can configure the devices that should display the time based entities in the Away Time Zone and the Away Time Zone's time. Two configuration fields are provided if you have several devices that are away and they are in different time zones.

10. NEW FEATURE - Log Activity for Zones - When a device exits a zone, the zone information (Enter Time, Exit Time, Time in the Zone (minutes & hours), zone name and device_name) can be written to a spreadsheet .csv file. This file can be imported into a spreadsheet (Excel, etc) and used for travel expense reporting, monitoring when a device entered and exited a zone, who entered a zone, etc. The zones and file name format is specified for each device you want to monitor. You can monitor multiple zones for one or more devices.

The file name format is:
    - 'zone_log-[year]-[zone]' to store the data by a specific zone for multiple devices. For example, you can keep records for Gary and Lillian in the Warehouse zone file.
    - 'zone_log-[year]-[device]' to store the data for a specific device for multiple zones. For example, you can keep records for Gary when he goes to the Warehouse and Storage Facility zones.
    - 'zone_log-[year]-[zone]-[device]' and 'zone_log-[year]-[device]-[zone]' to store the data for more than one device and more than one zone in the same file.  For example, you can keep records for Gary when he goes to the Warehouse zone in one file and Storage Facility zone in aqnother file. The two filename formats allow you to sort the spreadsheet file by device or by zone.

**Comment by amaisano** [comment]
Here is the log. Between 9am and 12pm on 12/1/2023 is when the funky behavior started. Around 12:15 PM is when I left the house and normal operations resumed.

[icloud3-event-log_2023.12.01-13.24.07.log.zip](https://github.com/gcobb321/icloud3_v3/files/13530883/icloud3-event-log_2023.12.01-13.24.07.log.zip)

**Comment by flemminglau** [comment]
> You need to select the FamIly Sharing List Device and the iOS App device (if the iOS App is installed on it) to add ít. Find my Friends devices are not available due to the Apple change.
> 
> A message should have been displayed when all the FamShr, FmF and iOS app devices are None.

Does this mean that it is still not working today (in rc8)? 

And that this fact is not mentioned anywhere but here?

**Comment by gcobb321** [comment]
@flemminglau 
The death of Find-my-Friends has been mentioned numerous times in numerous places. Click **Issues** at the top of this page and notice the pinned message on the top left side of the screen. That has been there since May, 2023. It's also been in the general comments and on the iCloud3 GitHub repository. 

Sorry but there is no fix for this in the near future.

**Comment by flemminglau** [comment]
Sorry I have not been able to find any mentions in my googling. 

And it seems to still be mentioned as a feature of icloud3 with no comments that it is currently actually not. 
Sorry. I am just frustrated that I discovered the problem the hard way trying to add the last of my needed devices.

**Comment by gcobb321** [comment]
Were you referring to the iCloud3 v2 docs or v3 docs. And yes, the v3 docs need to be updated. After rc9 comes out, I'll get back into the v3 docs.

**Comment by vajonam** [comment]
I cant get rid of the Event Log dialog that needs an update. I have tried everything. This is on an incognito browser. 

![image](https://github.com/gcobb321/icloud3_v3/assets/152501/a7e3d1ef-1846-404e-912f-e344ac6bc38a)

Any advice?

**Comment by gcobb321** [comment]
@vajonam 
What it’s happening is your browser has Event Log  v3.0.15 in its cache and is running that. iCloud3 has detected v3.0.16 in the _icloud3/event_log_ directory and has copied I it to the _config/www/icloud3_ or the directory it is being loaded from your configuration. As the message says, your browsers cache needs to be cleared so your browser will load the new version from your hard drive. 

I don’t know what you mean by an incognito browser or what browser you are running but Google ‘ how do I clear browser cachez’ for it and follow those directions. Once you have done that, refresh the screen to load the new version. It has been my experience that sometimes you have to do this several times to get it to refresh (Chrome) but it eventually will.

**Comment by vajonam** [comment]
Yes. I understand. I have cleared the cache. A private browser session or Incognito in google is a browser with an empty cache. I have had this in the past by with updates, but I cannot get to clear. When I click the refresh button on the browser event log it disappears

**Comment by gcobb321** [comment]
That may be but it is loading v3.0.15 from somewhere then. In the _config/www_ directory with the _icloud3-event-log-card.js_ file, open that .javascript file and verify the version. It is at the front of the file on a line like this:
` const version = "3.0.16" `. 

If it is 3.0.15, you may have a mismatch between the Lovelace Resource item for the _Event Log_ and the directory iCloud3 thinks it is being loaded from that displays in the _Event Log_ in Stage 1 of the start up routine. The directory is configured in _Configure Settings > Tracking and Other Parameters_ screen. It defaults to _www/icloud3_ in a new install of v3 but May be in _www/custom_cards_ if you ran v2. 

See the iCloud3 user Guide [here](https://gcobb321.github.io/icloud3_v3_docs/#/chapters/2.0-installing-and-configuring?id=step-53-create-the-event-log-portion-of-the-lovelace-screen) for more info about  iCloud3 and the Lovelace Resources directory. 

If you click _Event Log > Actions > _ , the version number running is displayed.

**Comment by vajonam** [comment]
Thanks that was it the I recopied over the `/www/icloud3/icloud3-event*.js`  to `/www/custom_cards/` and its resolved the issue. It was out of sync. Thanks for the quick pointers and help!

**Comment by ahamro** [comment]
I'm using iCloud3 for quite some time now but I still feel a beginner at it.
Now I want to install it from scratch and have to ask some really simple questions, for which I was not able to find the answers by searching the github pages.

1. Do I remove the 'orginal' iCloud integration before I install iCloud3?
2. Do I remove the Mobile App before I install iCloud3?
3. At the moment I have the iCloud3 v3 integration and the Mobile App running (not the iCloud integration). I don't see a way to add my Apple watches to iCloud3, because there are no entities for them.

**Comment by gcobb321** [comment]
@ahamro 
1. Remove it any time. Before or after doesn’t matter. 
2. Do not remove the Mobile App Integration.  It is responsible for the HA to iOS App and without it the iosapp tracking method will not work. 
3. Add the Watch on the _Configure Settings > Update Devices_ screen screen like any other device. iCloud3 controls the device tracker and sensor entities and adds them when you add a device you want to track in iCloud3. It also deletes them when you delete them from iCloud3

**Comment by ahamro** [comment]
Thanks Gary!

**Comment by ahamro** [comment]
When adding my wife's iPhone (which is in my Family Share list) I see the field "FAMILY SHARING LIST DEVICE - Use location data from this iCloud Acct Family Sharing member".
What do I enter in this field?

**Comment by gcobb321** [comment]
@ahamro 
Click on it to see a list of available famshr devices. See iCloud3 docs [here](https://gcobb321.github.io/icloud3_v3_docs/#/chapters/2-icloud-iosapp-loc-data?id=_21-location-data-icloud-account)

**Comment by amaisano** [comment]
Coming back to my issues with updates for the cloud items being incorrect, I've discovered a discrepancy between services.

When one clicks on the iC3 log widget an uses "Global Actions - Resume All Tracking", it results in this log entry:

`Service Action > Action-Resume All Tracking`

But when I use an automation and service call with the following config -- which I believe is set correctly to mimic the above command -- it results in a different log entry:

```YAML
service: icloud3.action
data:
  command: Resume Tracking
```

`Service Action > Action-Resume, Options-tracking`

The issue is, the Global Action > Resume All Tracking _correctly_ updates the location of my Cloud device (Apple Watch), while **the automation/service command does NOT**.

Maybe there is a missing config option in the service call? Why would these items not be the same log event? Are they calling the same exact code?

Thanks.

**Comment by jshep321** [comment]
Hi,
I think similar issues have been discussed but not this particular one.  Apologies if it's a duplicate.

When I arrive back to my home zone, the log shows an update immediately, which is perfect.  However, when I leave the home zone, there is quite a delay for the location to update.  Is it feasible from HA (or Node red preferably) to change the Next Update interval to a few seconds after a trigger (such as garage door closing, etc.)?   Changing the update interval to 1-2s while (zone change or X min timeout)  would be a fix (hack?) for any zone leaving delays.

**Comment by gcobb321** [comment]
@jshep321 
You can issue locate requests with the iCloud3 _action_ service call. See the iCloud3 docs [here](https://gcobb321.github.io/icloud3_v3_docs/#/chapters/7.4-service-calls?id=icloud3action) for more info.

**Comment by quixotic120** [comment]
just added and trying to install as per the install guide but cannot add the integration - get "this integration cannot be added from the UI" asking to manually add to config.yaml and link to documentation. Tried deleting and readding with HACS, relatively fresh HAOS install on RPI4, not sure what I am doing wrong?

**Comment by gcobb321** [comment]
@quixotic120 
It sounds like the v2 code is still installed on your system in the icloud3 directory, not the v3 code. 
1. Did you install from HACS in the icloud3 or the icloud3_v3 GitHub repository. If icloud3, did you have beta enabled in HACS? If not, you probably installed v2 again. 
2. Did you restart HA after installing from HACS?

**Comment by quixotic120** [comment]
@gcobb321 that was the issue, thanks! HACS was installing v2, had to check the beta updates slider.

**Comment by d13g0m0nt3s** [comment]
Hi, the last two days I’m having enormous battery drain on all my devices. Mostly at night, I wake up with my wife devices and mine with battery. The only change I made in my home is change my router, my ip range is 192.168.0.1 to 255 and before was 192.168.1.1 to 255. Could this affect to the iCloud3 configuration?. Should I reconfigure the integration ?

**Comment by cowboyrushforth** [comment]
Hello,  hope this is an OK place to ask this question.  

As I understand it, "gps" type of device tracker's take precedence when none of the non-gps trackers detect a device at home.

Previous to icloud v3, this issue did not occur on icloud v2.

here is what happens, screenshot attached from a recent day this happened, but it is still happening with 3.0.2.

Example user in this screenshot, they left the house at something like 8:30am. 

The router (The pfsense tracker), and an ios webhook from homekit have correctly told HA that the user is away.

Additionally the car (2022 Outback) gps tracker is updating and becomes the primary source of location, which seems to all work as I would expect.

Since Icloud v3 has not noticed the location, no big deal.  However a strange thing happens right before 9am.  Icloud v3 goes to update the location, and the user has arrived at another zone (Yoga).  

For a very short amount of time (like 5 seconds), the location is updated to Home, and then its updated again to Yoga.  The user was at Yoga the entire timespan in this screenshot.  

So in short:  when Icloud V3 catches up to other GPS trackers, somehow briefly the person entity is set to "home" right before its correctly set again to the zone the user is in.

Now this normally wouldn't be a big deal - but HA will run automations that notice this person is coming home, which is not ideal.

Any idea how to investigate this more?  Thanks for all the great software!
<img width="807" alt="Screenshot 2024-03-27 at 9 10 16 AM" src="https://github.com/gcobb321/icloud3_v3/assets/54122/c1d77d00-c6a0-4a0b-9d60-c467efc75030">

**Comment by gcobb321** [comment]
@cowboyrushforth 
iCloud3 relies on the Mobile App for zone exit triggers. If you are not using the Mobile App. set up an automation that runs the iCloud3 update > locate service call to do an immediate locate when one of the other trackers/sensors detect you have left Home or another zone. More info about this is in the iCloud3 docs > Service Calls chapter.

**Comment by Friedi1970** [comment]
Hi and thanks for this great integration which works great für my iPhones (where I installed the IOS app of HA). Anyway, I don't get my watch integrated because I can't find it anywhere since the FAMILY SHARING LIST DEVICE is empty.

Is there a way to add devices like watch, ipods and all those devices where no HA app is installed?

**Comment by JtwoA** [comment]
I think this is related to the issue I've got open, Gary.

**Comment by gcobb321** [comment]
@Friedi1970 @JtwoA 
 iCloud3 reads the devices you have set up in your iCloud Account Family Sharing List. If they are not set up ion the Family Sharing List or Location Sharing is not enabled, their information will not be returned to iCloud3.

If no devices are returned when iCloud3 starts, then it might not be set up or locations for the devices are not being shared.
- Go [here](https://support.apple.com/en-us/108380) to the Apple web site for info on setting up and checking your Family Sharing Account.
- Go to the Settings App on the iPhone or iPad. You should see Family under your picture. Make sure the family members are listed. Go to each one and make sure Location Sharing says they are sharing their location with you,
- Go to the FindMy App and make sure all the devices you want to track are listed on the devices screen.
- Go to the FindMy appon your iPhone. Go to the Me screen and make sure Share My location is enabled and From is 'This iPhone'.

Fix anything until it is correct.
- Go to the iCloud3 Configure > Update Devices screen. The devices should be listed in the FamShr Family Share selection field. Select the one that is tied to the iCloud3 device you are tracking.
-

**Comment by JtwoA** [comment]
Yep @gcobb321 - did these things.  And just for argument's sake, I opened my FIND MY app and every user and device is showing a current location undated "just now".

**Comment by gcobb321** [comment]
Then I have no idea why iCloud is not sending back any of your Family List device info.

Set the log level to raw data (Event Log > Actions > Show Raw Data.
Restart HA
Export the Event Log (Event Log > Actions > Export Event Log

Send me the icloud3-0.log file and the Event Log Export file so I can see what iCloud is returning. If the data is not being sent back. then there must be some setting somewhere on your iCloud account that is not correct.

**Comment by JtwoA** [comment]
Sent directly.

**Comment by Friedi1970** [comment]
@gcobb321 I did also all these things you mentioned. All the devices are listed in FindMy. It doesn't work; unfortunately.

**Comment by gcobb321** [comment]
@Friedi1970 
Go to _Configure Settings > Enter/Request Verification Code_ screen and request a new c code. Enter it and then restart iCloud3 with _Event Log > Actions > Restart iCloud3_ and see if anything is returned from iCloud.

**Comment by Friedi1970** [comment]
> @Friedi1970 Go to _Configure Settings > Enter/Request Verification Code_ screen and request a new c code. Enter it and then restart iCloud3 with _Event Log > Actions > Restart iCloud3_ and see if anything is returned from iCloud.

I requested as you described. The problem is I don’t get any code …

**Comment by gcobb321** [comment]
@Friedi1970 
Let’s take your problem out of the general question areas where I could lose track of it. Click the Bug icon at the top of the _Event Log_ and create a new issue.

**Comment by Friedi1970** [comment]
> Click the Bug icon at the top of the _Event Log_ and create a new issue.

What Event Log you are talking about?

**Comment by Kall0** [comment]
What a great integration! Been toying around with it and seems to be accuraty and faster than just using the mobile app and tracking person with that.

One question!
My son uses only Apple watch and ofc no mobile app. Any best practices to get the zone exit triggering faster other than just increasing the interval?
And another! Is there a setting that I just can't find or way to automate my self to set location unknown if X amount time sense last "located"?

**Comment by dgrnbrg** [comment]
Hello, I recently updated my iCloud3 to the latest stable version in HACS. Now, the sensor.icloud3_event_log doesn't exist any more. I have the recorder exclusion, and I can see that the device trackers are all working as normal.

This the only log that is relevant:

```
2024-08-09 17:51:40.681 WARNING (MainThread) [homeassistant.util.loop] Detected blocking call to open with args ('/config/.storage/icloud3/configuration',) inside the event loop by custom integration 'icloud3' at custom_components/icloud3/helpers/common.py, line 311: data = json_util.load_json(filename) (offender: /usr/src/homeassistant/homeassistant/util/json.py, line 74: with open(filename, mode="rb") as fdesc:), please create a bug report at https://github.com/gcobb321/icloud3_v3/issues
File "/config/custom_components/icloud3/__init__.py", line 71, in async_setup
config_file.load_storage_icloud3_configuration_file()
File "/config/custom_components/icloud3/support/config_file.py", line 68, in load_storage_icloud3_configuration_file
success = read_storage_icloud3_configuration_file()
File "/config/custom_components/icloud3/support/config_file.py", line 124, in read_storage_icloud3_configuration_file
File "/config/custom_components/icloud3/helpers/common.py", line 311, in load_json_file
```

besides the custom card errors due to the missing sensor:

```
TypeError: undefined is not an object (evaluating 'hass.states['sensor.icloud3_event_log'].attributes')
_btnClassMouseOut (/local/custom_cards/icloud3-event-log-card.js:1557:70)
/local/custom_cards/icloud3-event-log-card.js:766:83
```

**Comment by gcobb321** [comment]
@dgrnbrg  What iCloud3 version are you running? You can find the version number on the device _tracker attributes at the bottom. Also, what version of HA are you on?

**Comment by dgrnbrg** [comment]
Perhaps I might be mistaken that the trackers are working.

The tracker is just

```
source_type: gps
battery_level: 90
latitude: xx
longitude: xx 
gps_accuracy: 28
altitude: 2.6623884179893693
course: 104
speed: 2
vertical_accuracy: 2
friendly_name: xx-iPhone
```

I am on version 3.0.5.7 according to HACS, and HA version 2024.8.0 running on HAOS.

**Comment by gcobb321** [comment]
That tracker is from the Mobile App, not iCloud3.
You can get the attributes 2 ways:
1: Click on the device_tracker entity on a dashboard if you are displaying one, then click Attributes.
2. Go to the Sidebar > Developer Tools > State and filter by tracker.
    The iCloud3 device_tracker attributes is:
![attrs-device-tracker](https://github.com/user-attachments/assets/e03aad4f-6725-4b89-b58b-fbce01f84e7c)

**Comment by dgrnbrg** [comment]
I think that icloud3 isn't starting at all. Does that stack trace mean anything to you? It was working, and then I updated it via hacs, and now it doesn't work.

**Comment by gcobb321** [comment]
I looked at the stack track and the error it is showing at line 77 is correct, it is an error. I have fixed that problem in a future release. However, that code is only executed if you have a `platform: icloud3` line in your _configuration.yaml_ file somewhere and iCloud3 is also loading the old, non-integration way.

Remove the `platform: icloud3` statement and it will probably load.

**Comment by PeterLinuxOSS** [comment]
Hi, the icloud3 after every restart of HA it requires reconfiguration with error Authentication expired for iCloud3, even the the integration normally works with reconfiguration

**Comment by gcobb321** [comment]
@PeterLinuxOSS 
To reauthentication access to your Apple Account, go to the _Configure Settings > Enter/Request Verification Code_ screen and request a new 6-digit code, it will display on one of your devices. Then enter it and Submit.

**Comment by PeterLinuxOSS** [comment]
> @PeterLinuxOSS 
> To reauthentication access to your Apple Account, go to the _Configure Settings > Enter/Request Verification Code_ screen and request a new 6-digit code, it will display on one of your devices. Then enter it and Submit. 

But its working fine without the reauthentication, its just annoying it ask for it every restart when it's not needed.

**Comment by ivorunev** [comment]
Hello, I have the following problem: icloud v3 changes the name of the phone battery entity, for example, from "sensor.device_tracker_ivo_iphone_battery" to "sensor.device_tracker_ivo_iphone_battery_4" or other number, which prevents me from monitoring the battery level and managing a charger depending on it. What can I do to stop this and be able to see a gauge card without HA telling me the entity is missing ? Thank You in advance :)

**Comment by gcobb321** [comment]
HA creates the extension if it thinks the entity is still active when iCloud3 starts. I know that if you do a 'Reload'  iCloud3 instead of an HA Restart, the _4 type of extension is created.

**Comment by ivorunev** [comment]
> HA creates the extension if it thinks the entity is still active when iCloud3 starts. I know that if you do a 'Reload' iCloud3 instead of an HA Restart, the _4 type of extension is created.

Thanks man i'll try to avoid reloads in this case. Have a nice day 👍

**Comment by cedricdelecole** [comment]
Hello, 

I have just configured Icloud V3.1.3, with my apple account, and added the apple account of my daughter (which shows the her iphone mobile position), but also the apple account of my son, and here is the issue : its iphone position is not updated.

I have mentioned I have installed the HA mobile app on it, but it is not configured yet to connect to HA - but I believe ICloud should be able to track its position with the its apple account only.

Here are some screenshots showing the account is logged, and the device is normally setup to be tracked : 

Please note it uses the family, but I am not using that feature - just its own account : 

![image](https://github.com/user-attachments/assets/2a14ae3e-a629-49af-9bf2-8b860da0ded2)

The device is normally configured : 

![image](https://github.com/user-attachments/assets/20f2265f-1afb-4092-85ae-f25259ad7b45)


in here I must admit I am not sure about the options, especially the option about the zones, I am not sure what to put in there .... but I want, as soon as he is leaving home for example, that the device is being tracked, as well as even if he is in another zone ... Also I am not sure why the fact that there is the Mobile app installed on it needs to be mentioned there ?? What is this used for ?

![image](https://github.com/user-attachments/assets/8590fc23-90de-4375-b8e8-639a2ef132b6)

![image](https://github.com/user-attachments/assets/efe9209d-c6ac-4af9-8091-6349ab65489e)

Many thanks !

**Comment by gcobb321** [comment]
Closing due to no activity and the release of iCloud3 v3.1.5
